### PR TITLE
Render a more helpful error message in `Effect.timeout`

### PR DESCRIPTION
**Pull request by IMax153**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Related Issue #
- Closes #

### CODE REVIEW

**Comment by IMax153** [comment]
I suppose we could make this a patch release since I made the `duration` optional?

File: packages/effect/src/Cause.ts [hunk 1]
```diff
@@ -910,7 +911,10 @@ export const isRuntimeException: (u: unknown) => u is RuntimeException = core.is
  * @since 2.0.0
  * @category errors
  */
-export const TimeoutException: new(message?: string | undefined) => TimeoutException = core.TimeoutException
+export const TimeoutException: new(
+  duration?: Duration.Duration,
```
Comment (gcanti) [inline packages/effect/src/Cause.ts:915]: this is a breaking change, maybe

```ts
new(message?: string, duration?: Duration.Duration)
```

or

```ts
new(
  options?: string | {
    readonly message?: string
    readonly duration?: Duration.DurationInput
  }
)
```

Comment (IMax153) [reply packages/effect/src/Cause.ts:915]: Good call @gcanti - I'll opt for the second strategy for now unless someone else has strong opposition.

**Comment by IMax153** [comment]
Going to close and re-open a new PR given this is no longer a minor version bump.

**Pull request closed without merging**